ibis/expr/types/core.py

```diff
@@ -51,7 +51,7 @@ def __repr__(self) -> str:
             return repr(result)
 
     def __hash__(self) -> int:
-        return hash(self._key)
+        return hash(self._arg)
```

Comment:
The means that sticking and `Expr` and its `Node` into a hash table will trigger equality for or the other and fail for `Expr` and likely for `Node` as well.

Reply:
Originally I hashed the `self.__class__` as well to avoid that. Shall I put that back?

Reply:
Yeah, let's keep `__class__`